packages/where/index.js

```diff
@@ -59,32 +95,34 @@ export const whereEndoSock = (platform, env) => {
     } else {
       return raw`\\?\pipe\Endo\endo.pipe`;
     }
+  } else if (env.XDG_RUNTIME_DIR !== undefined) {
+    return `${env.XDG_RUNTIME_DIR}/endo/endo.sock`;
   } else if (platform === 'darwin') {
     if (env.HOME !== undefined) {
       return `${env.HOME}/Library/Application Support/Endo/endo.sock`;
     }
-  } else if (env.XDG_RUNTIME_DIR !== undefined) {
-    return `${env.XDG_RUNTIME_DIR}/endo/endo.sock`;
   } else if (env.USER !== undefined) {
     return `/tmp/endo-${env.USER}/endo.sock`;
   }
-  return 'endo.sock';
+  return 'endo/endo.sock';
 };
 
 /**
+ * Returns the most suitable path for Endo caches.
+ *
  * @type {typeof import('./types.js').whereEndoCache}
  */
 export const whereEndoCache = (platform, env) => {
-  if (platform === 'win32') {
-    return `${whereEndoStateWindows(env)}`;
+  if (env.XDG_CACHE_HOME !== undefined) {
+    return `${env.XDG_CACHE_HOME}/endo`;
+  } else if (platform === 'win32') {
+    return `${whereEndoHomeWindows(env)}`;
   } else if (platform === 'darwin') {
     if (env.HOME !== undefined) {
       return `${env.HOME}/Library/Caches/Endo`;
     }
-  } else if (env.XDG_CACHE_HOME !== undefined) {
-    return `${env.XDG_CACHE_HOME}/endo`;
   } else if (env.HOME !== undefined) {
     return `${env.HOME}/.cache/endo`;
   }
-  return '';
+  return 'endo/cache';
```

Comment:
likewise, this branch probably fades away; `$HOME` is always defined or definable

Reply:
then a possible concern would be "homeless" users, like service accounts that sometimes set their home to something `/` or `/dev/null`

Reply:
I’m going to willfully ignore the users without home directories. Endo’s a user agent. No user, no agent.